internal/pkg/aws/cloudformation/cloudformation.go

```diff
@@ -200,13 +200,32 @@ func (c *CloudFormation) Exists(name string) (bool, error) {
 	return true, nil
 }
 
-// Metadata returns the Metadata property of the CloudFormation stack's template.
-func (c *CloudFormation) Metadata(name string) (string, error) {
-	out, err := c.GetTemplateSummary(&cloudformation.GetTemplateSummaryInput{
-		StackName: aws.String(name),
-	})
+// MetadataOpts sets up optional parameters for Metadata function.
+type MetadataOpts func() *cloudformation.GetTemplateSummaryInput
+
+// MetadataWithStackName sets up the stack name for cloudformation.GetTemplateSummaryInput.
+func MetadataWithStackName(name string) MetadataOpts {
+	return func() *cloudformation.GetTemplateSummaryInput {
```

Comment:
Why do we want to return a function instead of just the `GetTemplateSummaryInput`? I am probably missing something - from the usage below it seems like it's fine to just return a `GetTemplateSummaryInput`?

Also I wonder if we need to add unit tests for these funcs?

Reply:
Good question! I think we can change it to not return a function. As for unit tests i think they don't have additional logics and they are tested in the unit tests for `Metadata()`